lib/Sema/TypeCheckPattern.cpp

```diff
@@ -838,6 +838,16 @@ bool TypeChecker::typeCheckParameterList(ParameterList *PL,
     
     checkTypeModifyingDeclAttributes(param);
     if (!hadError) {
+      // If we're in an enum case, then infer @_nonEphemeral for pointer
+      // payloads.
+      if (options.getBaseContext() == TypeResolverContext::EnumElementDecl) {
+        auto unwrappedType = type;
+        if (auto objectType = type->getOptionalObjectType())
+          unwrappedType = objectType;
+        if (unwrappedType->getAnyPointerElementType())
+          param->setNonEphemeral();
+      }
```

Comment:
Do we also want this for stored properties?

Reply:
Yes, and we're inferring `@_nonEphemeral` for parameters of memberwise initialisers – did you have another case in mind?

Reply:
I meant setters, sorry.

Reply:
Ah, I see! Though I can't immediately think of a case where you can use X-to-pointer conversions in conjunction with setters, AFAIK we only allow them for arguments of applies/subscripts.

Reply:
Ah, I didn't realize we didn't perform those conversions for assignments! Okay, good.

```swift
struct Test {
  var ptr: UnsafeRawPointer?
}

var x = Test()
x.ptr = "abc" // already an error
```